shared/fs/row/rows.js

```diff
@@ -19,6 +19,17 @@ type Props = {
   destinationPickerIndex?: number,
 }
 
+// can't just do `Types.RowItem & {key: string}` because flow doesn't like it.
+// TODO: maybe someday flow can figure it out?
+type RowItemWithKey =
+  | ({key: string} & Types.TlfTypeRowItem)
+  | ({key: string} & Types.TlfRowItem)
+  | ({key: string} & Types.StillRowItem)
+  | ({key: string} & Types.EditingRowItem)
+  | ({key: string} & Types.UploadingRowItem)
+  | ({key: string} & Types.PlaceholderRowItem)
+  | ({key: string} & Types.EmptyRowItem)
```

Comment:
That would work too; I guess I was sort of being lazy here, and that this would keep the key stuff in here instead of requiring a change in container. Although Nojima mentioned just using index would probably work here too, so I'm gonna try that. Thanks!